src/base/misc/fatfs.c

```diff
@@ -1637,6 +1637,7 @@ void mimic_boot_blk(void)
       loadaddress = SEGOFF2LINEAR(seg, ofs);
 
       LWORD(ebx) = f->drive_num;
+      LWORD(edx) = f->drive_num;
```

Comment:
Any reference to why do you think it improves
the compatibility with EDR-DOS kernels?
As far as I know, we do not support that DOS
yet, and if we did, any reason it would use the
`FD_D` case instead of its own one?
Have you managed to run one?

Reply:
A user could place EDR-DOS's DRBIO.SYS as KERNEL.SYS. As far as I know, EDR-DOS's protocol uses the FreeDOS boot sectors but may (unsure) use dl to pass the unit, and loads at 70h:0 instead of 60h:0.

It's not important I guess. And no, I haven't tried it yet. (Haven't implemented it in lDebug's boot loader yet either.)

Reply:
Then lets just skip it.
I usually don't apply patches when even
its author doesn't know why are they needed.